app/src/main/java/com/duckduckgo/app/statistics/pixels/Pixel.kt

```diff
@@ -262,30 +271,98 @@ interface Pixel {
     fun fire(pixel: PixelName, parameters: Map<String, String> = emptyMap(), encodedParameters: Map<String, String> = emptyMap())
     fun fire(pixelName: String, parameters: Map<String, String> = emptyMap(), encodedParameters: Map<String, String> = emptyMap())
     fun fireCompletable(pixelName: String, parameters: Map<String, String>, encodedParameters: Map<String, String> = emptyMap()): Completable
+    fun fireCompletable2(pixelName: String, parameters: Map<String, String>, encodedParameters: Map<String, String> = emptyMap()): Completable
 }
 
 class ApiBasedPixel @Inject constructor(
     private val api: PixelService,
     private val statisticsDataStore: StatisticsDataStore,
     private val variantManager: VariantManager,
-    private val deviceInfo: DeviceInfo
-) : Pixel {
+    private val deviceInfo: DeviceInfo,
+    private val pixelDao: PixelDao
+) : Pixel, LifecycleObserver {
+
+    private val compositeDisposable = CompositeDisposable()
+    private var syncJob: Job? = null
+
+    private val exceptionHandler: CoroutineExceptionHandler = CoroutineExceptionHandler { _, _ ->
+        Timber.i("pixel job failed exceptionHandler")
+    }
+
+    @InternalCoroutinesApi
+    @UiThread
+    @OnLifecycleEvent(Lifecycle.Event.ON_START)
+    fun onAppForegrounded() {
+        syncJob = GlobalScope.launch(exceptionHandler) {
+            pixelDao.pixels().collect { list ->
+                compositeDisposable.clear()
+                Timber.i("Pixel sending: $list")
+                list.forEach { pixelEntity ->
+                    compositeDisposable.add(
+                        fireApi(pixelEntity.pixelName, emptyMap(), emptyMap())
+                            .subscribeOn(Schedulers.io())
+                            .onErrorComplete()
+                            .subscribe({
+                                pixelDao.delete(pixelEntity)
+                                Timber.v("Pixel sent: ${pixelEntity.pixelName} with params: temp temp)")
+                            }, {
+                                Timber.w(it, "Pixel failed: ${pixelEntity.pixelName} with params:  temp temp)")
+                            })
+                    )
+                }
+
+            }
```

Comment:
Given that the API is already in RxJava land, why making the `PixelDao#pixels` return a `Flow`? this bit of code would read better if it would return just an `Observable` of the entity. What do you think?

Disclaimer: I know this PR is still WIP and I don't have yet much context in general but thought I'd drop some comments anyway. Feel free to disregard if necessary.

Reply:
Good point, yeah! I was tempted at the beginning to migrate every to coroutines and flow, but then discarded it to reduce some time. Yes, it has sense to keep this part using rx.

Keep those comments coming, actually, you saved me some time on the testing side throwing that comment before I start with them.